### PR TITLE
IDE-579 Add Crtl+N to the Repository Tree Tab to create new attribute

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixes IDE-579

Signed-off-by: dehilsterlexis david.dehilster@lexisnexis.com

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith ready for review.

File: wlib/RepositoryView.cpp [hunk 1]
```diff
@@ -61,6 +62,26 @@ BOOL CRepositoryView::PreTranslateMessage(MSG* pMsg)
...
+            {
+                TVITEM item;
+                item.mask = TVIF_HANDLE;
+                item.hItem = m_Tree.GetSelectedItem();
+                if (m_Tree.GetItem(&item))
+                {
+                    CComQIPtr<CModuleNode> modNode = (CTreeNode *)item.lParam;
+                    if (modNode) {
+                        CComQIPtr<CModuleNode> modNode = (CTreeNode *)item.lParam;
+                        if (modNode) {
```
Comment (GordonSmith) [inline wlib/RepositoryView.cpp]: To be sure, to be sure?

File: wlib/RepositoryTreeNode.h [hunk 2]
```diff
@@ -46,6 +46,8 @@ __interface IRepositorySlot
     void UpdateModule(IModule *from, IModule *to);
     IRepository * GetRepository(); 
     bool ExpandModule(IModule &mod);
+    bool DoInsertAttribute(CComPtr<IModule> module);
+
```
Comment (GordonSmith) [inline wlib/RepositoryTreeNode.h]: remove blank line

**Comment by dehilsterlexis** [comment]
@GordonSmith back to you

File: wlib/RepositoryView.cpp (hunk 1 above)
Comment (GordonSmith) [inline wlib/RepositoryView.cpp]: @dehilsterlexis This is still duplicate code?

**Comment by dehilsterlexis** [comment]
@GordonSmith Fixed the recursive duplication. Back to you.

**Comment by GordonSmith** [comment]
Looks good.

**Pull request merged**